src/prusti/all.ml

```diff
@@ -16,9 +15,36 @@ let rec all xs pred
    all(xs, p, res) ==> ens res=all(xs, p)
 @*)
 
+let rec repeat x n =
+  if n <= 0 then []
+  else x :: repeat x (n - 1)
+
+let is_positive x = x > 0
+
+let rec all_positive xs =
+  match xs with
+  | [] -> true
+  | x :: xs' -> x > 0 && all_positive xs'
+
+(* let has_property p xs = all p xs *)
+
+let test_all_positive n
+(*@ ex r ys; all_positive(ys, r); req res=ys; ens r=true/\res=ys @*)
```

Comment:
I added `req res=ys` because:

<details><summary>why3 file</summary>

```why
module M
  use int.Int
  use bool.Bool
  use list.List
  use all.Extras
  
  goal goal1:
    forall n : Int.int, res : List.list Int.int, v141 : List.list Int.int, v159 : 
          Bool.bool, v160 : Bool.bool, v161 : Int.int, v162 : List.list
                                                                Int.int, v184 : 
          Bool.bool, v194 : Bool.bool.
      ((((((v141 = v162) /\ (v194 = v159)) /\ (v160 = (Int.( > ) v161 0)))
           /\
           (v194 = v184)
           /\
           ((Int.( <= ) n 0) = false))
          /\
          (v184 = true)
          /\
          (res = (List.Cons 1 v141)))
         /\
         (((v141 = v162) /\ (v194 = v159)) /\ (v160 = (Int.( > ) v161 0)))
         /\
         (v194 = v184)
         /\
         ((Int.( <= ) n 0) = false))
        ->
        (exists r : Int.int.
           ((Bool.andb v160 v159) = true) /\ (res = (List.Cons v161 v162)))
end
```
</details> 

`v161 ` is universally quantified and the following counterexample by z3 shows the goal is false.

<details><summary>Counterexample</summary>

```
n : int = 1;
v159 : bool = true
v160 : bool = true
v161 : int = 17870
v184 :  bool = true
```
</details>

Reply:
That's a creative solution. I guess the intent is one of these:

```ml
(*@ ex r; all_positive(res, r); ens r=true @*)

(*@ all_positive(res, true) @*)
```

Unfortunately both don't work at the moment because of how we handle entailments stage by stage. I'm thinking about how to support it.

This works though, since all_positive doesn't have higher-order parameters and can be pure:

```ml
(*@ ens all_positive(res)=true @*)
```

I pushed here to show how to make the function available to Why3.